opentelemetry-resourcedetector-gcp/src/opentelemetry/resourcedetector/gcp_resource_detector/_gce.py

```diff
@@ -45,7 +45,7 @@ def host_type() -> str:
 
 
 def host_id() -> str:
-    return _metadata.get_metadata()["instance"]["id"]
+    return str(_metadata.get_metadata()["instance"]["id"])
```

Comment:
I don't have a google cloud instance at hand but are you sure this is not already a string?

Reply:
Yes it's a JSON int at least some of the time. It might return a string if it's too big for a JS integer
```
curl -s -H 'Content-Type: text/json' -H 'Metadata-Flavor:Google'  metadata/computeMetadata/v1/?recursive=true | jq '.instance.id'
2459451723172637643
```

Reply:
thanks for double checking